src/main/java/net/snowflake/ingest/streaming/internal/ClientConfigureRequest.java

```diff
@@ -0,0 +1,22 @@
+/*
+ * Copyright (c) 2024 Snowflake Computing Inc. All rights reserved.
+ */
+
+package net.snowflake.ingest.streaming.internal;
+
+/** Class used to serialize client configure request */
+class ClientConfigureRequest extends ConfigureRequest {
+  /**
+   * Constructor for client configure request
+   *
+   * @param role Role to be used for the request.
+   */
+  ClientConfigureRequest(String role) {
+    setRole(role);
```

Comment:
1. why not have the base class ctor accept a string role argument too?
2. Base class has role and file name as two fields, whereas this derived class doesn't add anything on top. So what are we gaining with this two-class-inheritance structure?

Reply:
It was existed to have two inheritors `ChannelConfigure` and `ClientConfigure`. Deleted for now and will check in when Iceberg feature merge.